pkg/e2e/autoscaler/autoscaler.go

```diff
@@ -178,13 +178,87 @@ func machineAutoscalerResource(targetMachineSet *mapiv1beta1.MachineSet, minRepl
 
 var _ = g.Describe("[Feature:Machines] Autoscaler should", func() {
 	defer g.GinkgoRecover()
+	scaleUpFunc := func(client runtimeclient.Client, targetMachineSet mapiv1beta1.MachineSet, nodeTestLabel string, initialNumberOfReplicas int32, expectedReplicas int32) {
+		g.By(fmt.Sprintf("Creating MachineAutoscaler object for targetMachineSet %q", targetMachineSet.Name))
```

Comment:
`scaleUpFunc` actually does not scale a machineset up. It just waits until a machineset is scaled down. Maybe `waitUntilScaledUpFunc`? Plus, `machineAutoscalerResource` does not scales up, it just annotates a machine set.